src/generator/AutoRest.AzureResourceSchema/ResourceSchemaParser.cs

```diff
@@ -82,6 +82,7 @@ public static class ResourceSchemaParser
                     JsonSchema resourceDefinition = new JsonSchema();
                     resourceDefinition.JsonType = "object";
 
+                    resourceDefinition.ResourceType = resourceType;
```

Comment:
This heuristic isn't reliable.  For example there are some swaggers that contain a property named "type" that is unrelated to resource type (I assume this is allowed).